### DISCUSSION TITLE
Installation on VPS - Internal Server Error

### DISCUSSION THREAD

**Discussion by xoomercom** [firefly-iii/firefly-iii#6750] [General]
I'm trying to install this great program on a VPS (knownhost)

I created a sub-domain, updated PHP to 8.0 and added missing extensions intl and sodium, the rest were there.  I used the softaculous installation v. 5.7.16 and everything goes through, the DB is created along with its user and I can see (via FTP) all the files that were installed in the firefly directory.

I proceed to the link http://abc.abc.com/firefly and get INTERNAL SERVER ERROR

I looked in the storage/logs folder and the log shows

[2022-12-27T15:51:27.418674-05:00] local.INFO: AUDIT: Show login form (1.1). (XX.XX.XX.XX -> GET:http://abc.abc.com/firefly/public/index.php/login) [] []

Please kindly help. Thanks much in advance.

**Comment by JC5** [comment]
There should be an error somewhere that explains the internal server error. Either in your dashboard or in the log directory in another log file. Without it, there's not much to say.

**Comment by xoomercom** [comment]
Thanks much for you reply.

This is what the hosting support came back with initially.


"
This looks to be a code issue in your application.

[Tue Dec 27 15:04:30.510567 2022] [http:error] [pid 12690] [client 100.XX.XX.XX:46665] AH02429: Response header name '[2022-12-27T15' contains invalid characters, aborting request
[Tue Dec 27 15:05:08.786418 2022] [http:error] [pid 12687] [client XX.XX.128.XXX:25445] AH02429: Response header name '[2022-12-27T15' contains invalid characters, aborting request

You'll need to check into what's giving that response and fix it.

"

**Reply by JC5** [reply]
They blame Firefly III but the error is an Apache related error. The error code "AH02429" is not from Firefly III.

But it does not matter, we can deal with shitty Apache servers. Set the following environment variables (don't ask me how though):

```
APP_LOG_LEVEL=emergency
AUDIT_LOG_LEVEL=emergency
```

That will probably stop the errors.

**Comment by xoomercom** [comment]
Thanks much.

Shared your response with the host, hope we can get it to work.

**Comment by xoomercom** [comment]
OK, so i got a response from the host, see below.

Hello,

This setting is not set on the Apache (server side) but within the .env file in the installation directory of the application itself.

I would recommend that any further issues be referenced to the developers.

As a one-time courtesy, I've made a backup of .env to .env.bak and updated the variables as requested:

---------------------------
root@XXXX [/home/XXXX/sub-domain/XXX/firefly]# diff .env .env-bak
57c57
< APP_LOG_LEVEL=emergency
---
> APP_LOG_LEVEL=notice
61c61
< AUDIT_LOG_LEVEL=emergency
---
> AUDIT_LOG_LEVEL=info
---------------------------


I now have access to the application...but...

**Comment by xoomercom** [comment]
I clicked around all the menus with no issues but when I go to edit or add a new asset account I get the following error.

[FireflyIII](http://XXX.XXX.com/firefly/public)
Whoops! An error occurred.

Unfortunately, this error was not recoverable :(. Firefly III broke. The error is:

Class "finfo" not found
More information

Please collect more information in the storage/logs directory where you will find log files. If you're running Docker, use docker logs -f [container]. You can read more about collecting error information in [the FAQ](https://docs.firefly-iii.org/faq/other#how-do-i-enable-debug-mode).
Get help [on GitHub](https://github.com/firefly-iii/firefly-iii/issues)

You're more than welcome to open a new issue on GitHub.

    Use the search!
    Include the information [from this debug page](http://abc.abc.com/firefly/public/debug).
    Tell us more than "it says Whoops!"
    Include error logs (see above).
    Tell us what you were doing.

**Comment by xoomercom** [comment]
edit: it actually happens when I try to add budget, expense account, bill, liabilities, .....

**Comment by xoomercom** [comment]
OK, did some googling and it would seem 'fileinfo' extension for PHP needs to be enabled...working on that now.

LMK if it's something else please.

**Reply by JC5** [reply]
That's it indeed.

**Comment by xoomercom** [comment]
Bingo! We are in business.  

Thanks for your help - hoping this helps somebody too.

BTW, App looks/works great!
